js/textInput.js

```diff
@@ -0,0 +1,120 @@
+const textarea = document.getElementById("textarea");
+
+// when the user click the button show the modal
+document.querySelector('#textarea-input').addEventListener('click', () => open_window())
+
+const open_window = () => {
+    clear_errors();
+    textarea.style.display = 'block'
+}
+
+// when the user click the x button close the pop-up
+const close_textarea = document.querySelector(".hide-textarea");
+close_textarea.onclick = function() {
+    textarea.style.display = "none";
+}
+
+// when the user click on the backdrop
+window.onclick = (event) => {
+    if (event.target === textarea)
+        textarea.style.display = "none";
+}
+
+
+function displayError(error) {
+    document.querySelector('#textarea-errors').innerHTML = `<h3 style="color: red">${error}</h3>`;
+}
+
+function clear_errors() {
+    document.querySelector('#textarea-errors').innerHTML = '';
+}
+
+function draw_nodes_from_text(nodesNum) {
+    let x = 100;
+    let y = 100;
+    let count = 0;
+
+    for (let i = 1; i <= nodesNum; i++) {
+        if (count <= 4) {
+            x += 100;
+            graph.addNode(new GraphPoint(x, y));
+            count++;
+        } else {
+            x = 100;
+            y += 100;
+            graph.addNode(new GraphPoint(x, y));
+            count = 0
+        }
+
+    }
+
+}
+
+function draw_edges_from_text(lines, nodesNum) {
+    for (let i = 1; i < lines.length; i++) {
+        // three case: 1. two nodes, two nodes and a weight, less than two nodes(error)
+        if (lines[i].length < 2) {
```

Comment:
javascript treat spaces as an element inside the lines array
for example : 
3 2
1 2
1 (there is space here)

it will work which but it should not

Reply:
I added the rule "start and end should be greater than 0"